mlir/utils/performance/parseMIOpenTuningDB.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+import sqlite3
+import sys
+
+# Useful tabled conversions
+MLIR_FILTER_LAYOUTS = {"NCHW": "kcyx", "NHWC": "kyxc"}
+MLIR_OUTPUT_LAYOUTS = {"NCHW": "nkhw", "NHWC": "nhwk"}
+MLIR_OPS = {"F": "conv2d", "W": "conv2d_bwd_weight", "B": "conv2d_bwd_data"}
+MLIR_DATA_TYPE = {"FP16": "f16", "FP32": "f32", "INT8INT8INT32": "i8"}
+
+# Convert a key,value pair from the tuning DB to a "--option value" parameter for rocmlir-gen
+def convertToRocmgenParam(key, val):
+    if key == "layout":
+        parsedName = f"--in_layout {val.lower()} --out_layout {MLIR_OUTPUT_LAYOUTS[val]} --fil_layout {MLIR_FILTER_LAYOUTS[val]} "
+    elif key in [
+        "in_w",
+        "in_h",
+        "in_channels",
+        "batchsize",
+        "fil_h",
+        "fil_w",
+        "out_channels",
+        "dilation_h",
+        "dilation_w",
+        "conv_stride_h",
+        "conv_stride_w",
+    ]:
+        parsedName = f"--{key} {val}"
+    elif key == "pad_w":
+        parsedName = f"--padding_w {val}"
+    elif key == "pad_h":
+        parsedName = f"--padding_h {val}"
+    elif key == "data_type":
+        parsedName = f"-t {MLIR_DATA_TYPE[val]}"
+    elif key == "direction":
+        parsedName = f"--operation {MLIR_OPS[val]}"
+    else:
+        return None
+
+    return parsedName
+
+
+def selectAllTables(cursor):
+    sql_query_get_all_tables = """SELECT name FROM sqlite_master WHERE type='table';"""
+    cursor.execute(sql_query_get_all_tables)
+    tables = cursor.fetchall()
+    return tables
+
+
+def selectAllConfigs(cursor):
+    sql_query_print_config = """SELECT * FROM config"""
```

Comment:
This will seriously limit the applicability of this script and make a normal db run into performance issues.